src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListenerBase.cs

```diff
@@ -227,18 +228,18 @@ private protected async virtual ValueTask ProcessWorkAsync(ImmutableArray<Work>
 
         cancellationToken.ThrowIfCancellationRequested();
 
-        // Early bail check for if we are disposed or somewhere in the middle of disposal 
+        // Early bail check for if we are disposed or somewhere in the middle of disposal
         if (_disposed || stream is null || solution is null)
```

Comment:
I just noticed that this class has a `_disposed` and a `_disposeTokenSource`. Is the extra `_disposed` field actually needed? Isn't `_disposeTokenSource.IsCanellationRequested` sufficient to check for disposal rather than having two bits of data?

Reply:
I removed disposed here and hopefully made this more clear.